### PR TITLE
New Error Handling lesson

### PR DESCRIPTION
**Pull request by doomspork**
Feedback welcomed.

### CODE REVIEW

File: lessons/advanced/error-handling.md [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+  |> File.read!
+rescue
+  e in KeyError -> IO.puts "missing :source_file option"
+  e in File.Error -> IO.puts "unable to read source file"
+end
+```
+
+## After
+
+Sometimes it may be necessary to perform some action after our `try/rescue`, regardless of whether an error occurrs or not. To handle this Elixir supports `try/after` which can be used inconjuntion with `try/rescue`.
```
Comment (doomspork) [inline lessons/advanced/error-handling.md]: occurrs

**Comment by doomspork** [comment]
@pjhampton on the topic of re-ordering lessons, what do you think about this one?  I think Error Handling should come before the OTP topics, thoughts?

**Comment by pjhampton** [comment]
This is good stuff! I just read through with no comments.

I think you are right. Running with the assumption that the reader is consuming the lessons from Basic - Specific, 1..n respectively, then this lessons position is Advanced 2 or 3 ~ I don't think which matters but definitely before OTP and Metaprogramming.

**Comment by doomspork** [comment]
Thanks @pjhampton, I went with Adv 2:

image

**Pull request merged**